### ISSUE TITLE
Just closes after one second

### ISSUE THREAD

**Issue by Philander** [nwoolls/MultiMiner#220]
Had an old version of multiminer that worked, but haven't used in a while, but was still opening fine.  After updating to latest version it closes as soon as it opens.  Tried older versions, didn't try extremely old versions only the last 5 or 6 to no avail.

Eventviewer "events" right after it closes.
-----------------
Faulting application name: System.ArgumentOutOfRangeException: Non-negative number required.
Parameter name: capacity

Server stack trace: 
   at MultiMiner.Utility.Net.IPRange.GetIPAddresses() in d:\Data\Visual Studio Projects\Personal\MultiMiner\MultiMiner\MultiMiner.Utility\Net\IPRange.cs:line 40
   at MultiMiner.Utility.Net.PortScanner.Find(IPAddress startingIp, IPAddress endingIp, Int32 startingPort, Int32 endingPort, Int32 connectTimeout) in d:\Data\Visual Studio Projects\Personal\MultiMiner\MultiMiner\MultiMiner.Utility\Net\PortScanner.cs:line 18
   at MultiMiner.Xgminer.Discovery.MinerFinder.Find(IPAddress startingIp, IPAddress endingIp, Int32 startingPort, Int32 endingPort) in d:\Data\Visual Studio Projects\Personal\MultiMiner\MultiMiner\MultiMiner.Xgminer.Discovery\MinerFinder.cs:line 16
   at MultiMiner.UX.ViewModels.ApplicationViewModel.FindNetworkDevices() in d:\Data\Visual Studio Projects\Personal\MultiMiner\MultiMiner\MultiMiner.UX\ViewModels\ApplicationViewModel.cs:line 1306
   at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
   at System.Runtime.Remoting.Messaging.StackBuilderSink.AsyncProcessMessage(IMessage msg, IMessageSink replySink)

Exception rethrown at [0]: 
   at System.Runtime.Remoting.Proxies.RealProxy.EndInvokeHelper(Message reqMsg, Boolean bProxyCase)
   at System.Runtime.Remoting.Proxies.RemotingProxy.Invoke(Object NotUsed, MessageData& msgData)
   at System.Action.EndInvoke(IAsyncResult result)
   at MultiMiner.UX.ViewModels.ApplicationViewModel.<>c__DisplayClass77.<FindNetworkDevicesAsync>b__75(IAsyncResult ar) in d:\Data\Visual Studio Projects\Personal\MultiMiner\MultiMiner\MultiMiner.UX\ViewModels\ApplicationViewModel.cs:line 1270
   at System.Runtime.Remoting.Messaging.AsyncResult.SyncProcessMessage(IMessage msg)
   at System.Runtime.Remoting.Messaging.StackBuilderSink.AsyncProcessMessage(IMessage msg, IMessageSink replySink)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch(), version: %2, time stamp: 0x%3
Faulting module name: %4, version: %5, time stamp: 0x%6
Exception code: 0x%7
Fault offset: 0x%8
Faulting process id: 0x%9
Faulting application start time: 0x%10
Faulting application path: %11
Faulting module path: %12
Report Id: %13
---------------------------------------------------------------------------------
--------------------------
Application: MultiMiner.Win.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ArgumentOutOfRangeException
Stack:

Server stack trace: 
   at MultiMiner.Utility.Net.IPRange.GetIPAddresses()
   at MultiMiner.Utility.Net.PortScanner.Find(IPAddress startingIp, IPAddress endingIp, Int32 startingPort, Int32 endingPort, Int32 connectTimeout)
   at MultiMiner.Xgminer.Discovery.MinerFinder.Find(IPAddress startingIp, IPAddress endingIp, Int32 startingPort, Int32 endingPort)
   at MultiMiner.UX.ViewModels.ApplicationViewModel.FindNetworkDevices()
   at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
   at System.Runtime.Remoting.Messaging.StackBuilderSink.AsyncProcessMessage(IMessage msg, IMessageSink replySink)
   at System.Runtime.Remoting.Proxies.RealProxy.EndInvokeHelper(System.Runtime.Remoting.Messaging.Message, Boolean)
   at System.Runtime.Remoting.Proxies.RemotingProxy.Invoke(System.Object, System.Runtime.Remoting.Proxies.MessageData ByRef)
   at System.Action.EndInvoke(System.IAsyncResult)
   at MultiMiner.UX.ViewModels.ApplicationViewModel+<>c__DisplayClass77.<FindNetworkDevicesAsync>b__75(System.IAsyncResult)
   at System.Runtime.Remoting.Messaging.AsyncResult.SyncProcessMessage(System.Runtime.Remoting.Messaging.IMessage)
   at System.Runtime.Remoting.Messaging.StackBuilderSink.AsyncProcessMessage(System.Runtime.Remoting.Messaging.IMessage, System.Runtime.Remoting.Messaging.IMessageSink)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
-----------------------------------------
----------------------------
Faulting application name: MultiMiner.Win.exe, version: 4.0.3.360, time stamp: 0x54fb0fa2
Faulting module name: KERNELBASE.dll, version: 6.1.7601.18409, time stamp: 0x5315a05a
Exception code: 0xe0434352
Fault offset: 0x000000000000940d
Faulting process id: 0x3d70
Faulting application start time: 0x01d0748d57113b58
Faulting application path: C:\Users\Petri\AppData\Local\MultiMiner\MultiMiner.Win.exe
Faulting module path: C:\Windows\system32\KERNELBASE.dll
Report Id: a0b50f5c-e080-11e4-bf87-ac9e174d3f88

**Comment by nwoolls** [comment]
Can you try this build?

https://www.dropbox.com/s/qm1h1u2y08r1x6y/MultiMiner-4.0.3.zip?dl=0

**Comment by Philander** [comment]
Hi, I resolved the issue.

Before I only did a repair to Dotnetfx4.5, but that didn't resolve the issue.  Today I uninstalled Dotnetfx4.5 and installed Dotnetfx4.0.  Multiminer works fine now.  After that I installed Dotnetfx4.5 again and Multiminer is still working. 

Unfortunately I didn't test the suggested build before the whole uninstall and installations of Dotnet since I only came on github now to post that it's working so I dont know if it would have worked without having to uninstall and install the dotnets.  Thank you though, I will keep it in case it happens again and report.
